lighthouse-core/config/v2/config.js

```diff
@@ -0,0 +1,182 @@
+const path = require('path');
+
+const defaultConfigPath = './default.json';
+const defaultConfigJson = require(defaultConfigPath);
+
+const _flatten = arr => [].concat(...arr);
+const _subtract = (setA, setB) => Array.from(setA).filter(x => !setB.has(x));
+
+class ConfigV2 {
+  constructor(configJson, configPath) {
+    if (!configJson) {
+      configJson = defaultConfigJson;
+      configPath = path.resolve(__dirname, defaultConfigPath);
+    }
+
+    // Perform a shallow clone so we can adjust gatherers and audits
+    configJson = Object.assign({}, configJson);
```

Comment:
I do shallow copies on any children I mutate and whenever this is exposed to outside for potential mutation it's parsed/stringified in `asJson`, this was mainly to preserve the ability to call from javascript with `audits: {custom: {implementation: MyAuditClass}}`